client/src/featureform/resources.py

```diff
@@ -1403,13 +1408,35 @@ class SparkCredentials:
     def __init__(self,
                  master: str,
                  deploy_mode: str,
+                 python_version: str,
                 ):
 
         if deploy_mode != "cluster" and deploy_mode != "client":
             raise Exception(f"Spark does not support '{deploy_mode}' deploy mode. It only supports 'cluster' and 'client'.")
 
+        python_version = self._verify_python_version(python_version)
+
         self.master = master
         self.deploy_mode = deploy_mode
+        self.python_version = python_version
+    
+    def _verify_python_version(self, version):
+        if version.count(".") == 2:
+            major, minor, _ = version.split(".")
+        elif version.count(".") == 1:
+            major, minor = version.split(".")    
+        else:
+            raise Exception("Please specify your Python version on the Spark cluster. Accepted formats: Major.Minor or Major.Minor.Patch; ex. '3.7' or '3.7.16")
+
+        if major != MAJOR_VERSION or minor not in MINOR_VERSIONS:
+            raise Exception(f"The Python version {version} is not supported. Currently, supported versions are 3.7-3.10.")
+        
+        if minor == "10":
```

Comment:
can you comment why this is here? Seems arbritrary

Reply:
So, on our docker image, we support 3.7 - 3.10 and this value will be used for `pyenv global {major}.{minor}.{patch}`. I can add a comment in the code too?

Reply:
yes please